### ISSUE TITLE
Batch insert with manual ObjectIds not working as documented

### ISSUE THREAD

**Issue by AslaugEir** [algolia/algoliasearch-client-scala#425]
According to [your documentation](https://www.algolia.com/doc/api-reference/api-methods/add-objects/?language=scala#add-objects-with-manual-objectid-assignments), I should be able to do 

```    
client.execute {
      batch (
        index into indexName objects items
      )
    } 
```
where `indexName` is a String and `items` is a Seq(MyScalaObjects), and MyScalaObjects has an encoder that serializes them to jsons.
However when I try sending in my objects in a Seq, I get the following error:
``` 
algolia.4XXAPIException: Failure "JSON root must be an object (array not allowed) near line:1 column:1", response status: 400
```

I am able to make the first example work (by adding a comma in the end of all but the last line, which is currently missing from your documentation)  
```     
client.execute {
      batch (
        index into indexName `object` items(0),
        index into indexName `object` items(1)
    }
``` 
so I am confident that there is not an issue with my object types, but sending in a Sequence of objects results in the error.

**Comment by ElPicador** [comment]
Hi there,

Batches are tested here: https://github.com/algolia/algoliasearch-client-scala/blob/master/src/test/scala/algolia/integration/DeleteByIntegrationTest.scala#L37-L39
And I can't reproduce with this test: https://github.com/algolia/algoliasearch-client-scala/pull/426/files

Could you share a bit more on your issue? The objects, the api version, etc.
If there is sensitive data feel free to send it to support@algolia.com.

Thanks!

**Comment by AslaugEir** [comment]
Version info:
User-Agent: Algolia for Scala (1.25.0); JVM (1.8.0_151); Scala (2.11.11)

Here is my object:
 ```
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
import io.circe.{Decoder, Encoder}


case class AlgoliaItem(
   objectId: Long,
   title: String,
   creatorId: Option[Int],
   creatorName: Option[String]
 )


 object AlgoliaItem{
  implicit val decode: Decoder[AlgoliaItem] = deriveDecoder
  implicit val encode: Encoder[AlgoliaItem] = deriveEncoder

 }
```

circe version is 0.9.0-M2. Like I said, this works for single objects at a time, so I don't think that is to blame, but including it for good measure.

And here is my code.

```
class AlgoliaProcessor(algoliaId: Option[String], algoliaKey: Option[String]) {
  private val algoliaClientId = algoliaId.getOrElse(sys.env.get("ALGOLIA_CLIENT_ID")).asInstanceOf[String]
  private val algoliaApiKey   = algoliaKey.getOrElse(sys.env.get("ALGOLIA_API_KEY")).asInstanceOf[String]
  val client                  = new AlgoliaClient(algoliaClientId, algoliaApiKey)
  def indexItems(indexName: String, items: Seq[AlgoliaItem]) = {
    client.execute {
      index into indexName `object` items
    }
  }
}
```

And the part where I'm calling it from
```
val itemsToIndex: Seq[AlgoliaItem] = itemStore.getItems(50, 50) 
algoliaProcessor.indexItems("foo_index", itemsToIndex)
```

I've simplified my objects and code as much as possible during debugging, but it still gives the same error.

**Comment by ElPicador** [comment]
The API uses json4s not circe for JSON (de)serialization. 

Honestly I've never tried it with circe. Could you also give me the dependencies you are using? So I try to reproduce it locally with circe.

**Comment by AslaugEir** [comment]
Ok, so I got it working. The mess is in the documentation. I ran all four possible combinations of using \`object\` vs objects and batching vs no batching, with the following results:

\`object\`, no batch: This should work according to your documentation, but does not work for me at least:
```
client.execute {
    index into indexName `object` items
}
```

objects, no batch: This is the way you do it in the test that you referred to. This works for me, but is not included in your documentation
```
client.execute {
    index into indexName objects items
}
```

objects, with batch: According to your documentation, this should not work, but it does.
```
client.execute {
    batch(
      index into indexName objects items
    )
}
```

And for good measure; \`object\`, no batch: Does not work for me, but then again, no-one said it would ;)
```
client.execute {
  batch(
    index into indexName `object` items
  )
}
```

So, basically, I am finding that the code works, just not as documented. You should maybe verify and see if you get same or different results, and update the documentation accordingly.

**Comment by ElPicador** [comment]
You are right, the doc was wrong.
I've fixed it. It should be deployed soon.

I'll also try to add more type safety to prevent this kind of mistakes.

**Comment by ElPicador** [comment]
The doc is deployed & fix. Thanks for finding this out!
